### PR TITLE
Migrate kube-proxy to use v1 Event API

### PR DESCRIPTION
**Pull request by yuanwoody**
#### What type of PR is this?
/kind feature

#### What this PR does / why we need it:
this migrate the kube-proxy to events.k8s.io v1

#### Which issue(s) this PR fixes:

Fixes #
Ref kubernetes/enhancements#383

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

**Comment by aojea** [comment]
/assign @thockin 
he may have more context on this

File: cmd/kube-proxy/app/server.go [hunk 1]
```diff
@@ -774,7 +775,7 @@ func (s *ProxyServer) Run() error {
 }
 
 func (s *ProxyServer) birthCry() {
-	s.Recorder.Eventf(s.NodeRef, api.EventTypeNormal, "Starting", "Starting kube-proxy.")
+	s.Recorder.Eventf(s.NodeRef, nil, api.EventTypeNormal, "Starting", "Starting Kube-proxy", "")
```
Comment (thockin) [inline cmd/kube-proxy/app/server.go:778]: we don't normally capitalize kube-proxy

File: pkg/proxy/util/utils.go [hunk 2]
```diff
@@ -267,7 +267,7 @@ func LogAndEmitIncorrectIPVersionEvent(recorder record.EventRecorder, fieldName,
 				Name:      svcName,
 				Namespace: svcNamespace,
 				UID:       svcUID,
-			}, v1.EventTypeWarning, "KubeProxyIncorrectIPVersion", errMsg)
+			}, nil, v1.EventTypeWarning, "KubeProxyIncorrectIPVersion", "Handle Endpoints", errMsg)
```
Comment (thockin) [inline pkg/proxy/util/utils.go:270]: Where does this string "Handle Endpoints" come from?  Should it be passed in to this func?

**Review by thockin — COMMENTED** [review]
really small points, thanks!

File: cmd/kube-proxy/app/server.go (hunk 1 above)
Comment (Yuan-Junliang) [reply cmd/kube-proxy/app/server.go:778]:
```suggestion
	s.Recorder.Eventf(s.NodeRef, nil, api.EventTypeNormal, "Starting", "Starting kube-proxy", "")
```

File: pkg/proxy/util/utils.go (hunk 2 above)
Comment (Yuan-Junliang) [reply pkg/proxy/util/utils.go:270]: I see that this function LogAndEmitIncorrectIPVersionEvent  is called to handle something of endpoints.
so I set the reason to "Handle Endpoints"
What do you think is appropriate to set the para reason?
thanks!

File: cmd/kube-proxy/app/server.go [hunk 3]
```diff
@@ -690,7 +691,7 @@ func (s *ProxyServer) Run() error {
 				// TODO(random-liu): Remove this when the docker bug is fixed.
 				const message = "CRI error: /sys is read-only: " +
 					"cannot modify conntrack limits, problems may arise later (If running Docker, see docker issue #24000)"
-				s.Recorder.Eventf(s.NodeRef, api.EventTypeWarning, err.Error(), message)
+				s.Recorder.Eventf(s.NodeRef, nil, api.EventTypeWarning, err.Error(), "Starting Kube-proxy", message)
```
Comment (Yuan-Junliang) [inline cmd/kube-proxy/app/server.go:694]:
```suggestion
				s.Recorder.Eventf(s.NodeRef, nil, api.EventTypeWarning, err.Error(), "Starting kube-proxy", message)
```

File: cmd/kube-proxy/app/server.go [hunk 4]
```diff
@@ -690,7 +691,7 @@ func (s *ProxyServer) Run() error {
 				// TODO(random-liu): Remove this when the docker bug is fixed.
 				const message = "CRI error: /sys is read-only: " +
 					"cannot modify conntrack limits, problems may arise later (If running Docker, see docker issue #24000)"
-				s.Recorder.Eventf(s.NodeRef, api.EventTypeWarning, err.Error(), message)
+				s.Recorder.Eventf(s.NodeRef, nil, api.EventTypeWarning, err.Error(), "Starting kube-proxy", message)
```
Comment (thockin) [inline cmd/kube-proxy/app/server.go:694]: StartProxy or StartKubeProxy

File: cmd/kube-proxy/app/server.go [hunk 5]
```diff
@@ -774,7 +775,7 @@ func (s *ProxyServer) Run() error {
 }
 
 func (s *ProxyServer) birthCry() {
-	s.Recorder.Eventf(s.NodeRef, api.EventTypeNormal, "Starting", "Starting kube-proxy.")
+	s.Recorder.Eventf(s.NodeRef, nil, api.EventTypeNormal, "Starting", "Starting kube-proxy", "")
```
Comment (thockin) [inline cmd/kube-proxy/app/server.go:778]: StartProxy or StartKubeProxy

File: pkg/proxy/util/utils.go (hunk 2 above)
Comment (Yuan-Junliang) [reply pkg/proxy/util/utils.go:270]:
```suggestion
			}, nil, v1.EventTypeWarning, "KubeProxyIncorrectIPVersion", "GatherEndpoints", errMsg)
```

File: pkg/proxy/ipvs/proxier.go [hunk 6]
```diff
@@ -1272,7 +1272,7 @@ func (proxier *Proxier) syncProxyRules() {
 								Name:      proxier.hostname,
 								UID:       types.UID(proxier.hostname),
 								Namespace: "",
-							}, v1.EventTypeWarning, err.Error(), msg)
+							}, nil, v1.EventTypeWarning, err.Error(), "Sync Proxy Rules", msg)
```
[the remark this answers is not in the document]
Comment (Yuan-Junliang) [reply pkg/proxy/ipvs/proxier.go:1275]:
```suggestion
							}, nil, v1.EventTypeWarning, err.Error(), "SyncProxyRules", msg)
```

File: pkg/proxy/iptables/proxier.go [hunk 7]
```diff
@@ -1125,7 +1125,7 @@ func (proxier *Proxier) syncProxyRules() {
 								Name:      proxier.hostname,
 								UID:       types.UID(proxier.hostname),
 								Namespace: "",
-							}, v1.EventTypeWarning, err.Error(), msg)
+							}, nil, v1.EventTypeWarning, err.Error(), "Sync Proxy Rules", msg)
```
[the remark this answers is not in the document]
Comment (Yuan-Junliang) [reply pkg/proxy/iptables/proxier.go:1128]:
```suggestion
							}, nil, v1.EventTypeWarning, err.Error(), "SyncProxyRules", msg)
```

File: pkg/proxy/healthcheck/service_health.go [hunk 8]
```diff
@@ -111,7 +111,7 @@ func (hcs *server) SyncServices(newServices map[types.NamespacedName]uint16) err
 						Namespace: nsn.Namespace,
 						Name:      nsn.Name,
 						UID:       types.UID(nsn.String()),
-					}, api.EventTypeWarning, "FailedToStartServiceHealthcheck", msg)
+					}, nil, api.EventTypeWarning, "FailedToStartServiceHealthcheck", "Health Check", msg)
```
[the remark this answers is not in the document]
Comment (Yuan-Junliang) [reply pkg/proxy/healthcheck/service_health.go:114]:
```suggestion
					}, nil, api.EventTypeWarning, "FailedToStartServiceHealthcheck", "Listen", msg)
```

File: pkg/proxy/healthcheck/proxier_health.go [hunk 9]
```diff
@@ -99,7 +99,7 @@ func (hs *proxierHealthServer) Run() error {
 		msg := fmt.Sprintf("failed to start proxier healthz on %s: %v", hs.addr, err)
 		// TODO(thockin): move eventing back to caller
 		if hs.recorder != nil {
-			hs.recorder.Eventf(hs.nodeRef, api.EventTypeWarning, "FailedToStartProxierHealthcheck", msg)
+			hs.recorder.Eventf(hs.nodeRef, nil, api.EventTypeWarning, "FailedToStartProxierHealthcheck", "Starting Kube-proxy", msg)
```
[the remark this answers is not in the document]
Comment (Yuan-Junliang) [reply pkg/proxy/healthcheck/proxier_health.go:102]:
```suggestion
			hs.recorder.Eventf(hs.nodeRef, nil, api.EventTypeWarning, "FailedToStartProxierHealthcheck", "StartKubeProxy", msg)
```

File: cmd/kube-proxy/app/server.go (hunk 4 above)
Comment (Yuan-Junliang) [reply cmd/kube-proxy/app/server.go:694]:
```suggestion
				s.Recorder.Eventf(s.NodeRef, nil, api.EventTypeWarning, err.Error(), "StartKubeProxy", message)
```

File: cmd/kube-proxy/app/server.go (hunk 5 above)
Comment (Yuan-Junliang) [reply cmd/kube-proxy/app/server.go:778]:
```suggestion
	s.Recorder.Eventf(s.NodeRef, nil, api.EventTypeNormal, "Starting", "StartKubeProxy", "")
```

**Review by thockin — COMMENTED** [review]
Thanks!

/lgtm
/approve

**Comment by thockin** [comment]
Yeah, I am not sure how you ended up with a merger commit, but my usual flow is:

```
git fetch upstream
git rebase upstream/master
```

**Comment by Yuan-Junliang** [comment]
@thockin 
Using git rebase, I made a mistake, this PR is something wrong.
So I create a new one #102832 .
please review it 
thanks!

**Pull request closed without merging**